src/Root.tsx

```diff
@@ -3,29 +3,45 @@ import { BrowserRouter as Router } from 'react-router-dom';
 import { ToastContainer } from 'react-toastify';
 
 import { CssBaseline } from '@mui/material';
-import { ThemeProvider } from '@mui/material/styles';
 
-import { theme } from '@graasp/ui';
+import { langs } from '@graasp/translations';
+import { ThemeProvider } from '@graasp/ui';
+
+import i18next from 'i18next';
 
 import App from './App';
-import i18nConfig from './config/i18n';
+import i18nConfig, { useCommonTranslation } from './config/i18n';
 import {
   QueryClientProvider,
   ReactQueryDevtools,
+  hooks,
   queryClient,
 } from './config/queryClient';
 
-const Root = (): JSX.Element => (
-  <QueryClientProvider client={queryClient}>
-    <I18nextProvider i18n={i18nConfig}>
-      <ThemeProvider theme={theme}>
-        <CssBaseline />
-        <ToastContainer stacked position="bottom-right" />
+const ThemeWrapper = () => {
+  const { i18n } = useCommonTranslation();
+  const { data: currentMember } = hooks.useCurrentMember();
+
+  return (
+    <ThemeProvider
+      langs={langs}
+      languageSelectSx={{ mb: 2, mr: 2 }}
+      i18n={i18n}
+      defaultDirection={i18next.dir(currentMember?.extra?.lang)}
```

Comment:
These can be replaced by using the `i18nConfig` object. 
```suggestion
      i18n={i18nConfig}
      defaultDirection={i18nConfig.dir(currentMember?.extra?.lang)}
```

See in the builder.